include/swift/AST/Decl.h

```diff
@@ -4086,13 +4089,17 @@ class VarDecl : public AbstractStorageDecl {
   TypeLoc getTypeLoc() const { return typeLoc; }
 
   bool hasType() const {
-    return !!typeInContext;
+    // We have a type if either the type has been computed already or if
+    // this is a deserialized declaration with an interface type.
+    return typeInContext ||
```

Comment:
I wanted to refactor things so that we never call getType() on deserialized ParamDecls, or VarDecls not in local context. Since VarDecls in local context that are not ParamDecls do not get seralized, this means we should be able to kill this hack altogether.